From e241ae78922c70c482a7f562b2a9879832989348 Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Sat, 21 Mar 2020 12:54:15 +0200
Subject: [PATCH] [SW-0] goya2: turn mem_cpy() weak

For Onix it should be override with NOP since PCIE PHY FW SRAM is not
implemented at FPGA model
An attempt to access somehow breaks later access to SRAM by ARC1

Change-Id: I2a4ac26b77df1f5c775e5a9b98b0e4093bd00c1c
---

diff --git a/common/mem_cpy.c b/common/mem_cpy.c
index 16e3d51..4de2056 100644
--- a/common/mem_cpy.c
+++ b/common/mem_cpy.c
@@ -8,7 +8,7 @@
 #include "arcboot.h"
 #include "mem_cpy.h"
 
-void mem_cpy(uint32_t image_size, uint64_t source, uint64_t dest)
+__attribute__((weak)) void mem_cpy(uint32_t image_size, uint64_t source, uint64_t dest)
 {
 	uint32_t mem_cpy_ctrl2;
 	uint16_t image_block_count_16b;
diff --git a/goya2_arc/onix.c b/goya2_arc/onix.c
index 211e27b..5b03968 100644
--- a/goya2_arc/onix.c
+++ b/goya2_arc/onix.c
@@ -27,3 +27,10 @@
 }
 
 void mem_cpy_wait_finish(void) {}
+void mem_cpy(uint32_t image_size, uint64_t source, uint64_t dest)
+{
+	(void)image_size;
+	(void)source;
+	(void)dest;
+}
+
